dynamic/main.go

```diff
@@ -85,7 +80,14 @@ func initialSetup() (tfbridge.ProviderInfo, pfbridge.ProviderMetadata, func() er
 			}
 
 			tfServer = p
-			defaultInfo.P.(*proto.Provider).Replace(p)
+			if tfServer != nil {
+				info = providerInfo(ctx, tfServer)
+			}
+
```

Comment:
Do we support re-parametrising the provider without restarting it?

Reply:
No, but there should probably be an explicit error there to signal that the parameterization was not successful.